Makefile

```diff
@@ -1,74 +1,116 @@
 SHELL ?= /bin/bash
 
-# The Docker registry where images are pushed.
-# Note that if you use an org (like on Quay and DockerHub), you should
-# include that: quay.io/foo
-DOCKER_REGISTRY    ?= brigadecore
+.DEFAULT_GOAL := build
 
-GIT_TAG   = $(shell git describe --tags --always 2>/dev/null)
-VERSION   ?= ${GIT_TAG}
-IMAGE_TAG ?= ${VERSION}
+################################################################################
+# Version details                                                              #
+################################################################################
 
-BINS = github-gateway check-run
-IMAGES = brigade-github-app brigade-github-check-run
+GIT_VERSION = $(shell git describe --always --abbrev=7 --dirty)
 
-.PHONY: build
-build: $(BINS)
+################################################################################
+# Go build details                                                             #
+################################################################################
 
-.PHONY: $(BINS)
-$(BINS): bootstrap
-	go build -o bin/$@ ./cmd/$@
+BASE_PACKAGE_NAME := github.com/brigadecore/brigade-github-app
 
-# Cross-compile for Docker+Linux
-.PHONY: build-docker-bins
-build-docker-bins: $(addsuffix -docker-bin,$(BINS))
+################################################################################
+# Containerized development environment-- or lack thereof                      #
+################################################################################
 
-%-docker-bin: bootstrap
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags '$(LDFLAGS)' -o ./rootfs/$* ./cmd/$*
+ifneq ($(SKIP_DOCKER),true)
+	DEV_IMAGE := quay.io/deis/lightweight-docker-go:v0.6.0
+	DOCKER_CMD := docker run \
+		-it \
+		--rm \
+		-e SKIP_DOCKER=true \
+		-v $$(pwd):/go/src/$(BASE_PACKAGE_NAME) \
+		-w /go/src/$(BASE_PACKAGE_NAME) $(DEV_IMAGE)
+endif
 
-# To use docker-build, you need to have Docker installed and configured. You should also set
-# DOCKER_REGISTRY to your own personal registry if you are not pushing to the official upstream.
-.PHONY: docker-build
-docker-build: build-docker-bins
-docker-build: $(addsuffix -image,$(IMAGES))
+################################################################################
+# Docker images we build and publish                                           #
+################################################################################
 
-%-image:
-	docker build -f Dockerfile.$* -t $(DOCKER_REGISTRY)/$*:$(IMAGE_TAG) .
+ifdef DOCKER_REGISTRY
+	DOCKER_REGISTRY := $(DOCKER_REGISTRY)/
+endif
 
-# You must be logged into DOCKER_REGISTRY before you can push.
-.PHONY: docker-push
-docker-push: $(addsuffix -push,$(IMAGES))
+ifdef DOCKER_ORG
```

Comment:
Checking in on the need for the separate `DOCKER_ORG` env var?  Would not `DOCKER_REGISTRY` suffice for all use cases?

`DOCKER_REGISTRY=quay.io/vdice make build`, `DOCKER_REGISTRY=vdice make build`, etc.

Reply:
It would _work_, but `quay.io/vdice` isn't a registry. `quay.io` is a registry and `vdice` is an org or user therein. One could say I am being a stickler here, but I found precedent for breaking them up in the project's existing secrets / `Brigade.js` where these were already defined separately as `project.secrets.dockerhubRegistry` and `project.secrets.dockerhubOrg` before I had even touched anything... so I just ran with that.